### PR TITLE
Change the IOperation unit test dump format for child nodes

### LINKED ISSUE #18752: [IOperation] Print lists on newlines
See https://github.com/dotnet/roslyn/pull/18698#discussion_r111781374. 

```
IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'Dim i1 As Integer')
  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'i1')
    Variables: Local_1: i1 As System.Int32
```
> This doesn't have to fixed right away, but the output would be more readable if the first item was printed on the new line. This applies to all lists.

### LINKED ISSUE #19866: Change the IOperation unit test dump format for child nodes
Currently, the dump prints the first child on the same line as the children header, which makes reviewing the dump very tedious. @AlekseyTs and @jcouv have both requested to change this format to dump the first child node on a new line.

It is a single line change in the dumper: we just need to append Environment.NewLine to the string argument passed to this LogString call: http://source.roslyn.io/#Roslyn.Test.Utilities/Compilation/OperationTreeVerifier.cs,245

However, this will need to update the expected output for almost all of the existing IOperation unit tests. We should do this sooner rather than later though.

@jinujoseph Can we get the vendors do this task as soon as possible?

### LINKED ISSUE #20194: Fix operation tree dumper to not print "static" for local function invocations
See https://github.com/dotnet/roslyn/pull/20177#discussion_r121803906

### PR DESCRIPTION
**Pull request by jinujoseph**
fixes #19866
Fixes #18752
Fixes #20194

Tagging @dotnet/analyzer-ioperation @dotnet/roslyn-compiler @cston for review.

### CODE REVIEW

**Comment by jinujoseph** [comment]
I am ok either way. I can wait for close to be merged PR's

**Comment by sharwell** [comment]
Fixes #18752

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs [hunk 1]
```diff
@@ -256,13 +268,14 @@ void M()
...
 ";
-            string expectedOperationTree = @"
+string expectedOperationTree = @"
 ILocalFunctionStatement (Local Function: void Foo()) (OperationKind.LocalFunctionStatement) (Syntax: 'void Foo() => x++;')
   IBlockStatement (2 statements) (OperationKind.BlockStatement) (Syntax: '=> x++')
     IExpressionStatement (OperationKind.ExpressionStatement) (Syntax: 'x++')
       IIncrementExpression (UnaryOperandKind.IntegerPostfixIncrement) (OperationKind.IncrementExpression, Type: System.Int32) (Syntax: 'x++')
         Left: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'x')
     IReturnStatement (OperationKind.ReturnStatement) (Syntax: '=> x++')
+    null
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs]: What does the trailing `"null"` represent?

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests.cs [hunk 2]
```diff
@@ -44,17 +44,29 @@ static void Test2(int y, params int[] x)
...
-      IConversionExpression (ConversionKind.Cast, Implicit) (OperationKind.ConversionExpression, Type: System.Int32[], Constant: null) (Syntax: 'null')
-        ILiteralExpression (Text: null) (OperationKind.LiteralExpression, Type: null, Constant: null) (Syntax: 'null')");
+@"IInvocationExpression (void Cls.Test1(params System.Int32[] x)) (OperationKind.InvocationExpression, Type: System.Void) (Syntax: 'Test1(null)')
+  Instance Receiver: null
+  Arguments(1):
+      IArgument (ArgumentKind.Explicit, Matching Parameter: x) (OperationKind.Argument) (Syntax: 'null')
+        IConversionExpression (ConversionKind.Cast, Implicit) (OperationKind.ConversionExpression, Type: System.Int32[], Constant: null) (Syntax: 'null')
+          ILiteralExpression (Text: null) (OperationKind.LiteralExpression, Type: null, Constant: null) (Syntax: 'null')
+        InConversion: null
+        OutConversion: null");
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests.cs]: Are `null` IOperation fields (such as `InConversion`) included explicitly now?

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IBlockStatement_MethodBlocks.vb [hunk 3]
```diff
@@ -18,15 +18,18 @@ Class Program
...
   IIfStatement (OperationKind.IfStatement) (Syntax: 'If 1 > 2 Th ... End If')
     Condition: IBinaryOperatorExpression (BinaryOperationKind.IntegerGreaterThan) (OperationKind.BinaryOperatorExpression, Type: System.Boolean, Constant: False) (Syntax: '1 > 2')
         Left: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
         Right: ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2) (Syntax: '2')
     IfTrue: IBlockStatement (0 statements) (OperationKind.BlockStatement) (Syntax: 'If 1 > 2 Th ... End If')
+    IfFalse: null
   ILabelStatement (Label: exit) (OperationKind.LabelStatement) (Syntax: 'End Sub')
+  null
   IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'End Sub')
+  null
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IBlockStatement_MethodBlocks.vb]: Why does the block contain `ILabelStatement` and `IReturnStatement`?

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs]: Is this `IReturnStatement.ReturnedValue`? If so, `OperationTreeVerifier.VisitReturnStatement` should probably visit the value explicitly with `Visit(operation.ReturnedValue, "Value");`.

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IBlockStatement_MethodBlocks.vb [hunk 4]
```diff
@@ -18,15 +18,18 @@ Class Program
...
+Dim expectedOperationTree = <![CDATA[
 IBlockStatement (3 statements) (OperationKind.BlockStatement) (Syntax: 'Sub Method( ... End Sub')
   IIfStatement (OperationKind.IfStatement) (Syntax: 'If 1 > 2 Th ... End If')
     Condition: IBinaryOperatorExpression (BinaryOperationKind.IntegerGreaterThan) (OperationKind.BinaryOperatorExpression, Type: System.Boolean, Constant: False) (Syntax: '1 > 2')
         Left: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
         Right: ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2) (Syntax: '2')
     IfTrue: IBlockStatement (0 statements) (OperationKind.BlockStatement) (Syntax: 'If 1 > 2 Th ... End If')
+    IfFalse: null
   ILabelStatement (Label: exit) (OperationKind.LabelStatement) (Syntax: 'End Sub')
+  null
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IBlockStatement_MethodBlocks.vb]: Is the `null` the `LabeledStatement`? If so, considering visiting that operation explicitly: `Visit(operation.LabeledStatement, "Statement");`.

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests.cs (hunk 2 above)
Comment (jinujoseph) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests.cs]: yes they are

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs (hunk 1 above)
Comment (jinujoseph) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs]: yes it's return value, i will address the feedback in the follow-up PR

**Comment by jinujoseph** [comment]
@cston , Merging this so to let others to use the same pattern. I will address your feedback as a follow-up PR.

**Pull request merged**